dev/plugins/hu.elte.txtuml.export.papyrus/src/hu/elte/txtuml/export/papyrus/wizardz/TxtUMLVisuzalizeWizard.java

```diff
@@ -67,106 +76,126 @@ public void addPages() {
 	 */
 	@Override
 	public boolean performFinish() {
-		String txtUMLModelName = selectTxtUmlPage.getTxtUmlModelClass();
-		Map<String, String> txtUMLLayout = selectTxtUmlPage.getTxtUmlLayout();
-		String txtUMLProjectName = selectTxtUmlPage.getTxtUmlProject();
-		String generatedFolderName = PreferencesManager
-				.getString(PreferencesManager.TXTUML_VISUALIZE_DESTINATION_FOLDER);
-
-		PreferencesManager.setValue(PreferencesManager.TXTUML_VISUALIZE_TXTUML_PROJECT, txtUMLProjectName);
-		PreferencesManager.setValue(PreferencesManager.TXTUML_VISUALIZE_TXTUML_MODEL, txtUMLModelName);
-		PreferencesManager.setValue(PreferencesManager.TXTUML_VISUALIZE_TXTUML_LAYOUT, txtUMLLayout.keySet());
-    
-		boolean saveSucceeded = SaveUtils.saveAffectedFiles(getShell(), txtUMLProjectName, txtUMLModelName, txtUMLLayout.keySet());
-		if (!saveSucceeded)
-			return false;
-
-		try {
-
-			this.checkEmptyLayoutDecsriptions();
-
-			IProgressService progressService = PlatformUI.getWorkbench().getProgressService();
-
-			progressService.runInUI(progressService, new IRunnableWithProgress() {
-				@Override
-				public void run(IProgressMonitor monitor) throws InterruptedException {
-					monitor.beginTask("Visualization", 100);
-
-					TxtUMLExporter exporter = new TxtUMLExporter(txtUMLProjectName, generatedFolderName,
-							txtUMLModelName, txtUMLLayout);
-					try {
-						exporter.cleanBeforeVisualization();
-					} catch (CoreException e) {
-						Dialogs.errorMsgb("txtUML export Error - cleaning resources",
-								"Error occured when cleaning resources.", e);
-						throw new InterruptedException();
-					}
-					monitor.subTask("Exporting txtUML Model to UML2 model...");
-					try {
-						TxtUMLToUML2.exportModel(txtUMLProjectName, txtUMLModelName,
-								txtUMLProjectName + "/" + generatedFolderName, ExportMode.ErrorHandlingNoActions, "gen");
-						monitor.worked(10);
-					} catch (Exception e) {
-						Dialogs.errorMsgb("txtUML export Error", "Error occured during the UML2 exportation.", e);
-						monitor.done();
-						throw new InterruptedException();
-					}
-
-					monitor.subTask("Generating txtUML layout description...");
-					TxtUMLLayoutDescriptor layoutDescriptor = null;
-					try {
-						layoutDescriptor = exporter.exportTxtUMLLayout();
+		List<IType> txtUMLLayout = selectTxtUmlPage.getTxtUmlLayout();
+		Map<Pair<String, String>, List<IType>> layoutConfigs = new HashMap<>();
+		for (IType layout : txtUMLLayout) {
+			IType innerLayoutClass = null;
+			try {
+				innerLayoutClass = Stream.of(layout.getTypes()).findFirst().get();
+			} catch (JavaModelException e) {
```

Comment:
This exception is thrown when the enclosing type (layout) is not present - removed after the layout selection and before the diagram generation. I don't believe it is possible, leastwise i cannot do that.